### PR TITLE
fetch: add edge case explanation for time drift

### PR DESCRIPTION
**Pull request by ZhouXing19**
Based on:

1. https://github.com/cockroachdb/molt/issues/93
2. https://cockroachlabs.slack.com/archives/C04TQ67V1JL/p1701871210926509

Release note: Updated the readme for a OS + docker based edge case with fetch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -245,6 +245,50 @@ molt fetch \
   --pg-logical-replication-slot-decoding 'pgoutput'
 ```
 
+### Edge case
+#### MscOS + CockroachDB as Source within Docker container
```
Comment (ryanluu12345) [inline README.md:249]: Typo: `MacOS` instead of `MscOS`.

**Comment by ZhouXing19** [comment]
@taroface tagging you for review as this may needed to be mentioned in doc too. The tricky part is that we don't have a solution to this so far.

File: README.md [hunk 2]
```diff
@@ -245,6 +245,50 @@ molt fetch \
...
+
+If you encountered an error similar to the following, please contact the support time and we will try to provide a solution.
+
+```
+ERROR: AS OF SYSTEM TIME: cannot specify timestamp in the future (1701836988.000000000,0 > 1701836987.322737000,0) (SQLSTATE XXUUU)
+```
+
+This error is due to the fact that with MacOS as the runtime OS, docker may have indeterministic time drift from the host machine.[[1](https://github.com/cockroachdb/molt/issues/93)]
```
Comment (ryanluu12345) [inline README.md:257]: Super nit: `Docker` instead of `docker`

**Review by ryanluu12345 — APPROVED** [review]
A few typos and wording changes. But otherwise LGTM. Merge after changes are in.

File: README.md [hunk 3]
```diff
@@ -245,6 +245,50 @@ molt fetch \
...
+time_diff_seconds=$(bc <<< "scale=6; $time_diff / 1000")
+
+echo "Time difference: ${time_diff} milliseconds"
+echo "Time difference: ${time_diff_seconds} seconds"
+echo
+
+done
+```
+
+This will cause problem when using `molt fetch` to export data from a CockroachDB within a container, as we run a `SELECT ... AS OF SYSTEM TIME` query to iterate content from the target table.
```
[the remark this answers is not in the document]
Comment (ZhouXing19) [reply README.md:290]: Good point, fixed it in https://github.com/cockroachdb/molt/pull/97. Thanks!!

**Pull request merged**